File: java/src/main/java/com/company/automation/models/ApiError.java
package com.company.automation.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * An error body.
 *
 * <p>The demo API is inconsistent: some endpoints return {@code {"message": "..."}} and others
 * {@code {"error": "..."}}. Both are modelled, and {@link #text()} returns whichever is present.
 *
 * <p>This is modelled as-is rather than normalised behind a single field, because a test asserting
 * on a tidied-up view would hide the inconsistency instead of documenting it. The inconsistency is
 * a finding worth raising with the API owners.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ApiError(
    @JsonProperty("message") String message, @JsonProperty("error") String error) {

  /** The human-readable text, whichever field carried it. Null when neither is present. */
  public String text() {
    if (message != null && !message.isBlank()) {
      return message;
    }
    if (error != null && !error.isBlank()) {
      return error;
    }
    return null;
  }
}

File: csharp/src/AutomationFramework.Core/Models/ApiError.cs
using System.Text.Json.Serialization;

namespace AutomationFramework.Core.Models;

/// <summary>
/// An error body.
/// </summary>
/// <remarks>
/// The demo API is inconsistent: some endpoints return <c>{"message": "..."}</c> and others
/// <c>{"error": "..."}</c>. Both are modelled, and <see cref="Text"/> returns whichever is present.
/// <para>
/// This is modelled as-is rather than normalised behind a single field, because a test asserting on a
/// tidied-up view would hide the inconsistency instead of documenting it. The inconsistency is a
/// finding worth raising with the API owners.
/// </para>
/// </remarks>
public sealed record ApiError(
    [property: JsonPropertyName("message")] string? Message,
    [property: JsonPropertyName("error")] string? Error)
{
    /// <summary>The human-readable text, whichever field carried it. Null when neither is present.</summary>
    public string? Text()
    {
        if (!string.IsNullOrWhiteSpace(Message))
        {
            return Message;
        }

        if (!string.IsNullOrWhiteSpace(Error))
        {
            return Error;
        }

        return null;
    }
}
